pinot-server/src/main/java/org/apache/pinot/server/starter/helix/SegmentOnlineOfflineStateModelFactory.java

```diff
@@ -91,6 +92,16 @@ public void onBecomeOnlineFromConsuming(Message message, NotificationContext con
 
       TableDataManager tableDataManager = _instanceDataManager.getTableDataManager(realtimeTableName);
       Preconditions.checkNotNull(tableDataManager);
+      if (tableDataManager instanceof RealtimeTableDataManager) {
```

Comment:
Please add the desired method in the `BaseTableDataManager` class with a default implementation that does nothing, and override it on the `RealtimeTableDataManager`.

Reply:
I can do this but I don't like the way the code ends up looking like: since we are talking about a TableDataManager and not a BaseTableDataManager, I will still have to cast TableDataManager to BaseTableDataManager before I invoke the method. (Note the hierarchy is as follows TableDataManager->BaseTableDataManager->RealTimeTableDataManager)  As an alternative I can bring these interfaces up to TableDataManager and override in BaseDataManager as nop function and then with the implementation I need in RealTimeTableDataManager. This has the problem that we start exposing low level interfaces in what should be the high level TableDataManager. So I think it is best to leave the code as is.

Reply:
You can add a method in TableDataManager interface with a default implementation.

Reply:
I will address as per our live discussion, see next commit for resolution